dataland-frontend/src/components/forms/CreateEUTaxonomyForNonFinancials.vue

```diff
@@ -1,139 +1,452 @@
+      LimitedAssurance: humanizeString("LimitedAssurance"),
+      ReasonableAssurance: humanizeString("ReasonableAssurance"),
     },
+
     postEuTaxonomyDataForNonFinancialsProcessed: false,
     messageCount: 0,
-    formInputsModel: {},
     postEuTaxonomyDataForNonFinancialsResponse: null,
     humanizeString: humanizeString,
   }),
+  watch: {
+    reportingPeriod: function (newValue: Date) {
+      this.reportingPeriodYear = newValue.getFullYear();
+    },
+  },
+  computed: {
+    namesOfFilesToUpload(): string[] {
+      const namesFromFilesToUpload = this.filesToUpload.map((el) => el.name.split(".")[0]);
+      const namesFromUploadedFiles = this.uploadFiles.map((el) => el.name.split(".")[0]);
+      return [...new Set([...namesFromFilesToUpload, ...namesFromUploadedFiles])];
+    },
+  },
   props: {
     companyID: {
       type: String,
     },
   },
+  mounted() {
+    const dataId = this.route.query.templateDataId;
+    if (typeof dataId === "string" && dataId !== "") {
+      this.editMode = true;
+      void this.loadEuData(dataId);
+    }
+  },
+
   methods: {
+    /**
+     * Loads the Dataset by the provided dataId and pre-configures the form to contain the data
+     * from the dataset
+     *
+     * @param dataId the id of the dataset to load
+     */
+    async loadEuData(dataId: string): Promise<void> {
+      this.waitingForData = true;
+      const euTaxonomyDataForNonFinancialsControllerApi = await new ApiClientProvider(
+        assertDefined(this.getKeycloakPromise)()
+      ).getEuTaxonomyDataForNonFinancialsControllerApi();
+
+      const dataResponse =
+        await euTaxonomyDataForNonFinancialsControllerApi.getCompanyAssociatedEuTaxonomyDataForNonFinancials(dataId);
+      const dataResponseData = dataResponse.data;
+      if (dataResponseData.data?.fiscalYearEnd) {
+        this.fiscalYearEnd = new Date(dataResponseData.data.fiscalYearEnd);
+      }
+      if (dataResponseData.data?.referencedReports) {
+        const propertiesOfFilesAssignedToDataID = dataResponseData.data.referencedReports;
+        for (const key in propertiesOfFilesAssignedToDataID) {
+          this.uploadFiles.push({
+            name: key,
+            reference: propertiesOfFilesAssignedToDataID[key].reference,
+            currency: propertiesOfFilesAssignedToDataID[key].currency,
+            reportDate: propertiesOfFilesAssignedToDataID[key].reportDate,
+            isGroupLevel: propertiesOfFilesAssignedToDataID[key].isGroupLevel,
+            convertedReportDate: propertiesOfFilesAssignedToDataID[key].reportDate
+              ? new Date(propertiesOfFilesAssignedToDataID[key].reportDate as string)
+              : "",
+          });
+        }
+      }
+      const receivedFormInputsModel = modifyObjectKeys(
+        JSON.parse(JSON.stringify(dataResponseData)) as objectType,
+        "receive"
+      );
+      this.waitingForData = false;
+      updateObject(this.formInputsModel, receivedFormInputsModel);
+    },
+
```

Comment:
Also in this area please adpot any changes you make in CreateEuTaxonomyFOrFinancials.vue.